pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1242,43 +1233,102 @@ func (vs *VSphere) CreateVolume(volumeOptions *VolumeOptions) (volumePath string
 		return "", err
 	}
 
-	// vmdks will be created inside kubevols directory
-	kubeVolsPath := filepath.Clean(ds.Path(VolDir)) + "/"
-	err = makeDirectoryInDatastore(vs.client, dc, kubeVolsPath, false)
-	if err != nil && err != ErrFileAlreadyExist {
-		glog.Errorf("Cannot create dir %#v. err %s", kubeVolsPath, err)
-		return "", err
-	}
-	glog.V(4).Infof("Created dir with path as %+q", kubeVolsPath)
+	// Create a disk with the VSAN storage capabilities specified in the volumeOptions.StorageProfileData.
+	// This is achieved by following steps:
+	// 1. Create dummy VM with the disk configured with VSAN policy attached.
+	// 2. Detach the disk from the dummy VM.
+	// 3. Delete the dummy VM. This removes all the vmx files.
+	// 4. Move the Virtual Disk to Kubevols directory.
+	// 5. Delete the dummy VM folder as now its contents are empty.
```

Comment:
Also, we would need one more call, detach the disk from dummy VM . As without detaching we can't attach it to the kubernetes node later on when a pod with this volume is deployed.